omegaconf/base.py

```diff
@@ -553,22 +553,34 @@ def _wrap_interpolation_result(
         throw_on_resolution_failure: bool,
     ) -> Optional["Node"]:
         from .basecontainer import BaseContainer
-        from .nodes import AnyNode
+        from .nodes import AnyNode, StringInterpolationResultNode
         from .omegaconf import _node_wrap
 
         assert parent is None or isinstance(parent, BaseContainer)
 
         if is_primitive_type(type(resolved)):
             # Primitive types get wrapped using `_node_wrap()`, ensuring value is
             # validated and potentially converted.
-            wrapped = _node_wrap(
-                type_=value._metadata.ref_type,
-                parent=parent,
-                is_optional=value._metadata.optional,
-                value=resolved,
-                key=key,
-                ref_type=value._metadata.ref_type,
-            )
+            # There is one exception when the following two conditions are met:
+            #   - `resolved` is a string, and
+            #   - the target type is either unspecified / Any / str
+            # In that case, we wrap `resolved` inside a `StringInterpolationResultNode`
+            # so as to ensure that it cannot be considered itself as an interpolation,
+            # e.g., when `resolved` would be a string like "${foo}".
+            target_type = value._metadata.ref_type
+            if isinstance(resolved, str) and target_type in [None, Any, str]:
+                wrapped: Node = StringInterpolationResultNode(
+                    value=resolved, key=key, parent=parent
+                )
+            else:
+                wrapped = _node_wrap(
+                    type_=target_type,
+                    parent=parent,
+                    is_optional=value._metadata.optional,
+                    value=resolved,
+                    key=key,
+                    ref_type=target_type,
+                )
```

Comment:
iirc, we use the resulting node for validation by calling validate_and_convert on it.

An idea is to have a TempNode that is delegating validate_and_convert to the function on the destination node.
This would allow us to potentially simplify here by always returning the same type from this function.
I am not 100% sure it's worth it though.

Reply:
This is even simpler than what I had in mind. If you can make it work like this I think it's better.

1. Your current logic is raising a validation error for container nodes, which I am not sure is correct.
2. At some point we added the validation logic for interpolation somewhere else, this should also be cleaned up.

Reply:
Ok I'll give it a shot a bit later today. Regarding your points:
1. I'll need to look into it more closely. Before this PR there was no proper validation for containers (see last point in this section: https://omegaconf.readthedocs.io/en/latest/structured_config.html#interpolations)
2. I'm not sure what you are referring to, can you be more specific? (before this PR the validation logic was from within `_node_wrap()`)

Reply:
1. Not validating is not the same as raising a validation error :)
2. If it was all localized to this function then I may be just misremembering or remembering an intermediate state.

Reply:
So, you were right regarding (2), the validation actually happened just before the call to `_node_wrap()`. That call was thus just to return a node of the correct type, and since we don't care about it anymore, this whole block can go away.

I didn't try to address the validation of containers (1) since I believe it's better left for a different PR.

Result is in b88350d